src/tests/Interop/IDynamicInterfaceCastable/Program.cs

```diff
@@ -149,6 +165,21 @@ public interface ITestGenericIntImpl: ITestGeneric<int, int>
         {
             return i;
         }
+
+        V ITestGeneric<int, int>.DoubleGenericArg<V>(V v)
+        {
+            if (v is int i)
+            {
+                i *= 2;
+                return Unsafe.As<int, V>(ref i);
+            }
+            else if (v is string s)
+            {
```

Comment:
Can you also validate that `typeof(V) == typeof(string)` here?It  will make ensure that the hidden generic arg is correct.